Flex based reflow of website for mobile

klanmiko opened the pull request:
* Landing takes up entire screen height
* Use flex to center landing content
* Set root element font size to 18px on large screens, 16 on medium. Recalculate em's to maintain current sizing
* Center social good illustration on mobile and desktop
* tracks icons appear next to the header on mobile, are of fixed size
* sponsor text is now as big as FAQ text, not awkwardly bigger
* our logo takes up the top of the landing on mobile
* center landing content at 492px. Makes sense as it means the empty space on the right of the HackDavis title is less than 4/5 of title's width at this breakpoint, approximately.
* section padding from 100px horizontal to 50px horizontal on small screens
  *  Change the stats row margins from negative to zero, so that the green lines still appear

RIP my Tuesday

klanmiko commented on assets/css/styles.css:
```diff
@@ -468,18 +507,20 @@ img.social-good {
 
 h2.tracks-title {
   line-height: 1.25;
+  font-size: 1.7rem;
```
2rem to 1.7rem

lovincyrus commented:
Is this ready for review or are you still making changes?

lovincyrus commented:
750px breaks, gotcha. Which mobile size should I test on? iPhone 6/7/8? @klanmiko

klanmiko commented:
> Is this ready for review or are you still making changes? @klanmiko

You viewed it in the middle of the changes I was making, had to open it up on chrome.

> 750px breaks, gotcha. Which mobile size should I test on? iPhone 6/7/8? @klanmiko

I meant I was moving stuff from a 750px media query into the big 768px one haha

>     1. landing illustration's `transform-origin` is off
That is true. But I have also used flex with the transform origin to make sure that the tower does not scale down into the upper left corner, it now scales towards the bottom right, near center.

>     2. huge gap between tracks illustration with the track's title - refer to attachement
Fixed by [36a8746](https://github.com/HackDavis/website-2019/pull/5/commits/36a8746748593c7b79f063ef5f5f36c80a8f37e5)
>     3. schedule's css is messed up - refer to attachment
Cannot reproduce, maybe I fixed it?
>     4. increased `font-size` of content (too large), not recommended
To explain my font size changes, the landing page text is larger than it is before, such that the landing text is the same size as the about text, which is the same size as the sponsors text. Before my changes, about text was at 18 while the baseline was 16. I found that the about text looked slightly off, and I would argue that standardizing the font size makes more sense. I've changed the font scaling so it is 18px on lg, 16px on md, and 14px on sm

> The only changes that made sense to me were:
> 
>     1. center social good container
> 
>     2. reduced stats margin
These were the two of the goals I had when I started. I wanted to also be able to scale down the font size on mobile, which wasn't there so I redid the font-sizing on some of the elements, and center most elements while maintaining consistent margins on mobile (FAQ and sponsors text was sticking out)

lovincyrus commented on assets/css/styles.css:
```diff
@@ -341,33 +348,43 @@ button.sponsor-us {
+  -moz-transform-origin : 70% 90%;
+  -o-transform-origin : 70% 90%;
+  -ms-transform-origin : 70% 90%;
+  transform-origin : 70% 90%;
```
```suggestion
  transform-origin : 70% 60%;
```

Changed:
`70% 60%`'s position makes more sense, otherwise it will stuck at the top right corner at 1440px

lovincyrus replied:
Refer to the review, otherwise resolved.

klanmiko commented on index.html:
```diff
@@ -147,35 +146,41 @@ <h1 class="social-good-text" data-entrance="fade">
+          <div class="col-md-4 readable-text-col" data-entrance="fade" data-entrance-delay="100">
+            <div class="tracks-header">
+              <div class="tracks-picture">
+                <img src="assets/img/heart.svg" alt="heart" class="tracks-img">
```
On second thought it doesn't look that good

klanmiko commented on assets/css/styles.css:
```diff
@@ -309,8 +315,8 @@ img.hackdavis {
 }
 
```
I would do 15px padding-top, seems like too big of a gap

lovincyrus replied:
yeah but what about the gap between pavilion and btns

padding can take care of the inconsistent gaps:

<img width="398" alt="screen shot 2018-11-14 at 8 58 53 pm" src="https://user-images.githubusercontent.com/1021101/48531289-8c14c800-e850-11e8-87d4-90c338576db2.png">

klanmiko replied:
Space between HackDavis and details is 42 px, space between code and february is ~18-19px. And the space between buttons and the pavillion is 50px. 

I still think that it looks awkward having that much space. With 15px it is the height of another <br>

lovincyrus commented on assets/css/styles.css:
```diff
@@ -379,11 +379,11 @@ img.tower {
+  -moz-transform-origin : 70% 16vw;
+  -o-transform-origin : 70% 16vw;
+  -ms-transform-origin : 70% 16vw;
+  transform-origin : 70% 16vw;
```
What's up with the inconsistent % and vw

Stick with one, please. I suggested 70% 60% too.

klanmiko replied:
yea, broke on MBP

lovincyrus commented on assets/css/styles.css:
```diff
@@ -547,7 +580,7 @@ p.tracks-des {
 }
 
```
Before:
<img width="312" alt="screen shot 2018-11-14 at 9 53 00 pm" src="https://user-images.githubusercontent.com/1021101/48533049-cdf53c80-e857-11e8-9cf6-65a23df66a47.png">


After:
```
.time-box {
    transform: translate(-65px);
}
```

klanmiko replied:
it's now flush on all screen sizes

lovincyrus commented on assets/css/styles.css:
```diff
@@ -585,18 +585,19 @@ p.tracks-des {
+	-webkit-transform: translate(-60px, -5px);
+  -moz-transform: translate(-60px, -5px);
+  -o-transform: translate(-60px, -5px);
+	transform: translate(-60px, -5px);
```
translate(-60px,1px)
